src/cpucounters.cpp

```diff
@@ -1110,8 +1110,6 @@ bool PCM::discoverSystemTopology()
     };
     std::unordered_map<int, domain> topologyDomainMap;
     {
-        TemporalThreadAffinity aff0(0);
```

Comment:
If we remove this line it is possible the execution of this function can be suspended on one core and resumed on other core by the OS. On hybrid system with P and E-cores I am not sure if this will run into a rare sporadic behavior which we never tested. I would prefer to pin the affinity to ref core 0.

Reply:
Since the ref core array is not initialized yet, you might try to pin to core 0..n until it does not fail.

Reply:
this should be easy to implement adding try/catch in this scope with incrementing core id